### ISSUE TITLE
Неправильная конвертация символа ñ (испанский->русский)

### ISSUE THREAD

**Issue by Winand** [BladeMight/Mahou#347]
Mahou 2.9.0.1 d2f32d6
Конфигурация [Mahou.ini.gz](https://github.com/BladeMight/Mahou/files/6968725/Mahou.ini.gz)

Я использую следующую конфигурацию языков (чтобы по Alt-Tab переключать только две раскладки):

- русский
- английский
  - раскладка США
  - раскладка Испанская

![image](https://user-images.githubusercontent.com/53390/129037187-93a3f405-af9c-4e63-98a6-39deee6ff12a.png)

Проблема состоит в том, что конвертация с раскладкой США проходит верно, а из испанской нет
1. jkl; -> олд**ж**
2. jkl**ñ** -> олд**ё**

**Comment by Winand** [comment]
Да.. OEM клавиши у разных раскладок в разных местах находятся [ES](https://kbdlayout.info/KBDSP/scancodes), [US](https://kbdlayout.info/kbdusx/scancodes)

**Comment by Winand** [comment]
Коды меняются только при переводе из испанской. А обратно? Сейчас так: º -> ё -> ñ -> ж

**Comment by BladeMight** [comment]
![esp](https://user-images.githubusercontent.com/11931888/129099078-d50b7bca-068e-40d0-ac7d-0ead24d12791.gif)

**Comment by Winand** [comment]
![hog](https://user-images.githubusercontent.com/53390/129101229-2d942404-a485-47f8-8eba-f30c94f1dde6.gif)

Содержимое `LayoutKeyCodeReplace.txt`
```>1034
[221]=[187]
[192]=[186],[186]=[219],[219]=[189],[189]=[191],[191]=[220],[220]=[192]
<1034
```

**Comment by BladeMight** [comment]
@Winand Ну не знаю... У меня похожим образом работает если вообще **убрать** испанскую раскладку из системы, но оставить в настройках Mahou.ini при включенном "Переключать между раскладками" и одной из раскладок Испанск.

Попробуй Включить "Переключать между раскладками" и поставь 2 раскладки в Испанскую и Русскую, применить и проверить.

**Comment by BladeMight** [comment]
Вот почти полностью со стандартными настройками даже работает, только включил `Переключать между раскладками` и раскладки 1 и 2 настроил на рус./исп. (хотя у меня он и так работает нормально):
[bundle.zip](https://github.com/BladeMight/Mahou/files/6971383/bundle.zip)

**Comment by Winand** [comment]
Да, теперь я понимаю. У меня такие настройки:
![image](https://user-images.githubusercontent.com/53390/129103335-71471772-cf3d-4655-9be8-ed7362a466c9.png)

Но весь смысл в том, чтобы переключаться и между англ/рус и между исп/рус. В зависимости от того, какую из иностранных раскладок я выбрал последней.

**Comment by BladeMight** [comment]
> Но весь смысл в том, чтобы переключаться и между англ/рус и между исп/рус. В зависимости от того, какую из иностранных раскладок я выбрал последней.

Но Mahou види ВСЕ раскладки поэтому он будет переключать между всеми если эта галочка не включена, и у меня так тоже работает нормально, т.к. эти правила подмены кодов срабатывают когда раскладка меняется "НА" и "С" раскладки 1034, т.е. испанской.

**Comment by Winand** [comment]
Если эмулируется Alt+Shift, как я понимаю, переключение происходит так же как в Windows. У меня специально добавлен _один_ иностранный язык с _двумя_ раскладками, чтобы переключение происходило между русской и _одной_ иностранной раскладкой.
Увы, в mahou выбирается жёстко только две раскладки, и я обхожу это через флажок эмуляции Alt-Shift.

**Comment by BladeMight** [comment]
> Увы, в mahou выбирается жёстко только две раскладки, и я обхожу это через флажок эмуляции Alt-Shift.

Неплохой вариант. Можно также поставить 2 основные и добавить одно сочетание клавиш для смены 3-ю.
